### PR TITLE
PRISM-11343 - Wrap fingerprint implementation within a timeout

### PR DESCRIPTION
**Pull request by precious-ossai-cko**
## Issue

https://checkout.atlassian.net/browse/PRISM-11343

## Proposed changes

Wrap fingerprint implementation within a timeout

## Test Steps

> 1. On the `fingerprintTimeoutInterval`, change the `timeoutInterval` to 0.1.
> 2. Use an example app to run the application, it should log the timeout error
> 3. Revert the value after testing.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [x] Reviewers assigned
* [x] I have performed a self-review of my code and manual testing
* [x] Lint and unit tests pass locally with my changes
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] I have added necessary documentation (if applicable)

### CODE REVIEW

File: Sources/Risk/Risk.swift [hunk 1]
```diff
@@ -41,22 +45,41 @@ public final class Risk {
...
+                self.loggerService.log(riskEvent: .publishFailure, blockTime: self.blockTime, deviceDataPersistTime: nil, fpLoadTime: self.fingerprintService?.fpLoadTime, fpPublishTime: nil, deviceSessionId: nil, requestId: nil, error: RiskLogError(reason: "publishData", message: RiskError.Publish.fingerprintTimeout.localizedDescription, status: nil, type: "Timeout"))
+                completion(.failure(.fingerprintTimeout))
+          }
+        }
+
+        fingerprintService?.publishData { [weak self] fpResult in
+          guard let self = self else { return }
+
+          DispatchQueue.main.async {
+            guard let timer = self.timer, timer.isValid else { // 2.59 -> valid
```
Comment (okhan-okbay-cko) [inline Sources/Risk/Risk.swift:67]: Timer would become invalid at `2.59` seconds. We can remove these comments hence they are the things that I forgot to delete :)

File: Sources/Risk/Risk.swift [hunk 2]
```diff
@@ -41,22 +45,41 @@ public final class Risk {
         }
     }
     
-    public func publishData (cardToken: String? = nil, completion: @escaping (Result<PublishRiskData, RiskError.Publish>) -> Void) {
-        guard let fingerprintService = fingerprintService else {
-            completion(.failure(.fingerprintServiceIsNotConfigured))
-            return
+    public func publishData(cardToken: String? = nil, completion: @escaping (Result<PublishRiskData, RiskError.Publish>) -> Void) {
+        guard let _ = fingerprintService else {
```
Comment (okhan-okbay-cko) [inline Sources/Risk/Risk.swift:49]: This is another oversight. We can bring this unwrapping back and use the non optional version on line 63 below

**Review by okhan-okbay-cko — COMMENTED** [review]
Can we add some unit tests around the newly added code please?

File: Tests/Mocks/MockRisk.swift [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//
+//  File.swift
+//  
+//
+//  Created by Precious Ossai on 08/07/2024.
+//
+import XCTest
+@testable import Risk
+
+import Foundation
```
Comment (okhan-okbay-cko) [inline Tests/Mocks/MockRisk.swift:10]: Is this not used anymore?

File: Tests/RiskTests/RiskTests.swift [hunk 4]
```diff
@@ -10,38 +10,141 @@ import XCTest
...
-          expectation.fulfill()
+            XCTAssertNotNil(error)
+            expectation.fulfill()
         }
-
+        
         waitForExpectations(timeout: 5, handler: nil)
     }
+    
+    //    func testPublishTimeout() {
```
Comment (okhan-okbay-cko) [inline Tests/RiskTests/RiskTests.swift:63]: we can remove this completely

File: Tests/RiskTests/RiskTests.swift [hunk 5]
```diff
@@ -10,38 +10,141 @@ import XCTest
...
-    //         createdRiskInstance = risk
-    //         expectation.fulfill()
-    //     }
-
-    //     waitForExpectations(timeout: 5, handler: nil)
-    //     XCTAssertNotNil(createdRiskInstance)
-    // }
-
+    
+//    func testGetInstanceWithValidConfiguration() {
```
Comment (okhan-okbay-cko) [inline Tests/RiskTests/RiskTests.swift:14]: also this wouldn't help anymore. We can keep this commented out code somewhere else internally, if we think we'll need it

File: Tests/Mocks/MockRisk.swift (hunk 3 above)
Comment (precious-ossai-cko) [reply Tests/Mocks/MockRisk.swift]: No,  removed

**Review by okhan-okbay-cko — APPROVED** [review]

**Review by okhan-okbay-cko — APPROVED** [review]
🚢

**Review by ehab-al-cko — APPROVED** [review]

**Pull request merged**